core-plugins/src/e2e-test/features/joiner/Joiner.feature

```diff
@@ -75,5 +75,5 @@ Feature: Joiner analytics - Verify File source data transfer using Joiner analyt
     Then Wait till pipeline is in running state
     Then Open and capture logs
     Then Verify the pipeline status is "Succeeded"
-    Then Verify the CSV Output File matches the Expected Output File With Expected Partition
+    Then Verify the CSV Output File matches the Expected Output File: "joinerOutput" With Expected Partition: "expectedJoinerOutputPartitions"
```

Comment:
why are we making changes to joiner test here, seems unrelated to error collector?

Reply:
I have move the `verifyOutput` logic to a same place under [CorePlugin.java](https://github.com/cdapio/hydrator-plugins/blob/38f130df787f004360d3069286faae812395516a/core-plugins/src/e2e-test/java/io/cdap/plugin/common/stepsdesign/CorePlugin.java) so it can be reused by `ErrorCollector`. Since the partition of `ErrorCollector` is always 1 but partition of `Joiner` can change, I introduced another parameter `partition` to the function, which then changes the original `Joiner` feature step.